tools/sr-snapshots/Sources/Files/Directory.swift

```diff
@@ -0,0 +1,96 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// Enables operations on real directory in file system.
+public struct Directory: DirectoryProtocol {
```

Comment:
⚠️ There is a visible room for improvement in implementation of some methods of this construct (e.g. transactions or I/O failure resilience). I kept it bare simple and quite naive. If seeking for more excellence, I'd rather switch to using 3rd party (e.g. [JohnSundell/Files](https://github.com/JohnSundell/Files)) for most operations in this file.